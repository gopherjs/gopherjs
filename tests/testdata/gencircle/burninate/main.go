// Test of instances of generic types inverted dependencies.
// The `burnable` imports `dragons` but the instance of `Trogdor` requires
// `burnable`. This is a simple check that the all packages are loaded before
// the types finish setting up. This is similar to the "simple" gencircle test
// except with generic functions and methods.
package main

import (
	"github.com/gopherjs/gopherjs/tests/testdata/gencircle/burninate/burnable"
	"github.com/gopherjs/gopherjs/tests/testdata/gencircle/burninate/dragon"
)

func main() {
	d := dragon.Trogdor[burnable.Cottages]{}
	b := burnable.Cottages{}
	burnable.Burn(d, b)
}
