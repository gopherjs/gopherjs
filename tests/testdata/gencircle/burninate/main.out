burninating the thatched-roof cottages
